src/app/pages/applications/forms/chart-form.component.ts

```diff
@@ -0,0 +1,343 @@
+      case 'string':
+        type = 'input';
+        if (schema.enum) {
+          type = 'select';
+        }
+        break;
+      case 'boolean':
+        type = 'checkbox';
+        break;
+      case 'list':
+        type = 'list';
+        break;
+      case 'hostpath':
+        type = 'explorer';
+        break;
+      default:
+        type = 'input';
+    }
+
+    return type;
+  }
+
+  getSelectOptions(schema) {
+    const options = [];
+    schema.enum.forEach(option => {
+      options.push({
+        value: option.value,
+        label: option.description,
+      });
+    });
+    return options;
+  }
+
+  getFieldConfigs(config, parent=null) {
+    let fieldConfigs = [];
+    let name = config.variable;
+    if (parent) {
+      name = `${parent.variable}_${name}`;
+    }
+    const fieldConfig = {
+      type: this.getType(config.schema),
+      required: config.schema.required,
+      value: config.schema.default,
+      tooltip: config.description,
+      placeholder: config.label,
+      name: name,
+    }
+
+    if (fieldConfig.type == 'select') {
+      fieldConfig['options'] = this.getSelectOptions(config.schema);
+    }
+
+    if (config.schema.private) {
+      fieldConfig['togglePw'] = true;
+    }
+
+    if (fieldConfig.type == 'explorer') {
+      fieldConfig['explorerType'] = 'directory';
+      fieldConfig['initial'] = '/mnt';
+    }
+
+    if (fieldConfig.type == 'list') {
+      const listFields = this.getTemplateListField(config, parent);
+      fieldConfig['templateListField'] = listFields;
+      fieldConfig['box'] = true;
+      fieldConfig['listFields'] = [];
+    }
+
+    fieldConfigs.push(fieldConfig);
+
+    if (config.schema.subquestions) {
+      config.schema.subquestions.forEach(subquestion => {
+        let sbu_name = subquestion.variable;
+        if (parent) {
+          sbu_name = `${parent.variable}_${sbu_name}`;
+        }
+        const subFieldConfig = {
+          type: this.getType(subquestion.schema),
+          name: sbu_name,
+          placeholder: config.label,
+        }
+
+        if (config.schema.show_subquestions_if) {
+          subFieldConfig['isHidden'] = true;
+          subFieldConfig['relation'] = [{
+            action: 'SHOW',
+              when: [{
+                name: name,
+                value: true,
+              }]
+          }];
+        }
+
+        if (subFieldConfig.type == 'explorer') {
+          subFieldConfig['explorerType'] = 'directory';
+          subFieldConfig['initial'] = '/mnt';
+        }
+
+        fieldConfigs.push(subFieldConfig);
+      });
+    }
+
+    return fieldConfigs;
+  }
+
+  getTemplateListField(config, parent) {
+    let listFields = [];
+    config.schema.items.forEach(item => {
+      const fields = this.getFieldConfigs(item, parent);
+      listFields = listFields.concat(fields);
+    });
+    return listFields;
+  }
+
+  parseSchema(catalogApp) {
+    this.catalogApp = catalogApp;
+    this.title = this.catalogApp.name; 
```

Comment:
Needly done! 
One edgecase: This function does not seem to handle mistakes in question.yaml very gracefully.
Very much an edgecase, because the CI is solid. But it could happen (certainly if the community starts to get more involved).
Maybe it's nice to just throw an error to the user?
Something simple like: "Error detected reading App"